src/daidepp/keywords.py

```diff
@@ -0,0 +1,541 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import List, Optional, Union
+
+from daidepp.constants import *
+
+
+@dataclass
+class Unit:
+    power: POWER
+    unit_type: UNIT_TYPE
+    province: PROVINCE
+
+    def __str__(self):
+        return f"{self.power} {self.unit_type} {self.province}"
+
+
+@dataclass
+class HLD:
+    unit: Unit
+
+    def __str__(self):
+        return f"({self.unit}) HLD"
+
+
+@dataclass
+class MTO:
+    unit: Unit
+    province: PROVINCE
+
+    def __str__(self):
+        return f"({self.unit}) MTO {self.province}"
+
+
+@dataclass
+class SUP:
+    unit_1: Unit
+    unit_2: Unit
+    province_no_coast: Optional[PROVINCE_NO_COAST] = None
+
+    def __str__(self):
+        if not self.province_no_coast:
+            return f"({self.unit_1}) SUP ({self.unit_2})"
+        else:
+            return f"({self.unit_1}) SUP ({self.unit_2}) MTO {self.province_no_coast}"
+
+
+@dataclass
+class CVY:
+    unit_1: Unit
+    unit_2: Unit
+    province: PROVINCE
+
+    def __str__(self):
+        return f"({self.unit_1}) CVY ({self.unit_2}) CTO {self.province}"
+
+
+@dataclass
+class MoveByCVY:
+    unit: Unit
+    province: PROVINCE
+    province_seas: List[PROVINCE_SEA]
+
+    def __init__(self, unit, province, *province_seas):
+        self.unit = unit
+        self.province = province
+        self.province_seas = province_seas
+
+    def __str__(self):
+        return (
+            f"({self.unit}) CTO {self.province} VIA ("
+            + " ".join(self.province_seas)
+            + ")"
+        )
+
+
+@dataclass
+class RTO:
+    unit: Unit
+    province: PROVINCE
+
+    def __str__(self):
+        return f"({self.unit}) RTO {self.province}"
+
+
+@dataclass
+class DSB:
+    unit: Unit
+
+    def __str__(self):
+        return f"({self.unit}) DSB"
+
+
+@dataclass
+class BLD:
+    unit: Unit
+
+    def __str__(self):
+        return f"({self.unit}) BLD"
+
+
+@dataclass
+class REM:
+    unit: Unit
+
+    def __str__(self):
+        return f"({self.unit}) REM"
+
+
+@dataclass
+class WVE:
+    power: POWER
+
+    def __str__(self):
+        return f"{self.power} WVE"
+
+
+@dataclass
+class Turn:
+    season: SEASON
+    year: int
+
+    def __str__(self):
+        return f"{self.season} {self.year}"
+
+
+@dataclass
+class PCE:
+    powers: List[POWER]
+
+    def __init__(self, *powers):
+        self.powers = powers
+
+    def __str__(self):
+        return "PCE (" + " ".join(self.powers) + ")"
```

Comment:
I think all parentheses should have a whitespace character separating them from the following token.

Reply:
So you're saying instead of something like `PRP (PCE (AUS ENG))`, it should be `PRP ( PCE ( AUS ENG ))` ?

Reply:
Yup `PRP ( PCE ( AUS ENG ) )`. That's how all Albert bot messages are formatted.